components/claim/components/email_to_ldap.tsx

```diff
@@ -11,81 +10,103 @@ import {t} from 'utils/i18n.jsx';
 import LoginMfa from 'components/login/login_mfa.jsx';
 import LocalizedInput from 'components/localized_input/localized_input';
 
-export default class EmailToLDAP extends React.PureComponent {
-    static propTypes = {
-        email: PropTypes.string,
-        siteName: PropTypes.string,
-        ldapLoginFieldName: PropTypes.string,
-    };
+type Props = {
+    email: string;
+    siteName?: string;
+    ldapLoginFieldName?: string;
+};
+
+type State = {
+    passwordError: string;
+    ldapError: string;
+    ldapPasswordError: string;
+    serverError: string;
+    showMfa: boolean;
+    ldapId: string;
+    password: string;
+    ldapPassword: string;
+};
 
-    constructor(props) {
+export default class EmailToLDAP extends React.PureComponent <Props, State> {
+    private emailPasswordInput: React.RefObject<HTMLInputElement>;
+    private ldapIdInput: React.RefObject<HTMLInputElement>;
+    private ldapPasswordInput: React.RefObject<HTMLInputElement>;
+
+    public constructor(props: Props) {
         super(props);
 
+        this.emailPasswordInput = React.createRef();
+        this.ldapIdInput = React.createRef();
+        this.ldapPasswordInput = React.createRef();
         this.state = {
             passwordError: '',
             ldapError: '',
             ldapPasswordError: '',
             serverError: '',
             showMfa: false,
+            ldapId: '',
+            password: '',
+            ldapPassword: '',
         };
-
-        this.emailPasswordInput = React.createRef();
-        this.ldapIdInput = React.createRef();
-        this.ldapPasswordInput = React.createRef();
     }
 
-    preSubmit = (e) => {
+    public preSubmit = (e: React.SyntheticEvent): void => {
         e.preventDefault();
 
-        var state = {
+        const state = {
             passwordError: '',
             ldapError: '',
             ldapPasswordError: '',
             serverError: '',
+            password: '',
+            ldapId: '',
+            ldapPassword: '',
         };
 
+        const defaultState = state;
```

Comment:
Instead of creating a separate `defaultState`, couldn't you just use `state` below?